### ISSUE TITLE
Ability to clear search

### ISSUE THREAD

**Issue by 91khr** [VSCodeVim/Vim#4108]
> English is not my native language, please excuse typing errors and grammar mistakes.

I think the highlighted searching results that I no longer need are annoying.
Is there a way to clear the search patterns like following vim command does?

```vim
let @/=''
```

**Comment by J-Fields** [comment]
Does `:noh` do what you're asking for?

**Comment by 91khr** [comment]
> 
> 
> Does `:noh` do what you're asking for?

No, I've read #1053. If I use `:nohl`, I have to turn on it manually before my next search.
And `:nohl` only turns off the highlighting, but the search pattern is still there and can be used by `n`.

Now my alternative is `/\\\\`(searching a pattern that will never exist).

**Comment by J-Fields** [comment]
> If I use :nohl, I have to turn on it manually before my next search.

This isn't true - `:nohl` only disables it until a new search is initiated.

> And :nohl only turns off the highlighting, but the search pattern is still there and can be used by n

This is true. Seems like we should support `let @/=...` for this use case.

**Comment by 91khr** [comment]
Thanks. It seems that I've confused `:nohl` with `:set nohl`.

**Comment by xconverge** [comment]
An example to remap it is in the README too

```
  "vim.normalModeKeyBindingsNonRecursive": [
    {
      "before": ["<C-n>"],
      "commands": [":nohl"]
    }
  ],
```

**Comment by Chiel92** [comment]
The following seems to work better than the suggestions so far. By also specifying after things like cancel multiple cursors keep working.
```
        {
          "before":["<Esc>"],
          "after":["<Esc>"],
          "commands": [
            ":nohl"
          ]
        }
```

**Comment by diminutivesloop** [comment]
It would be great if there was an extension command for clearing search so I could bind it to Escape when vim is in normal mode. I can't use `vim.normalModeKeyBindingsNonRecursive` for this because I've intentionally disabled the default binding for `extension.vim_escape` in normal mode since it interferes with built-in VSCode functionality (https://github.com/VSCodeVim/Vim/issues/9459).
